packages/astro/src/runtime/server/hydration.ts

```diff
@@ -15,21 +15,31 @@ export interface HydrationMetadata {
 	componentExport: { value: string };
 }
 
+type Props = Record<string | number | symbol, any>;
+
 interface ExtractedProps {
 	isPage: boolean;
 	hydration: HydrationMetadata | null;
-	props: Record<string | number | symbol, any>;
+	props: Props;
 }
 
 const transitionDirectivesToCopyOnIsland = Object.freeze([
 	'data-astro-transition-scope',
 	'data-astro-transition-persist',
 ]);
 
+export function withoutTransitionAttributes(props: Props): Props {
+	const res: Props = Object.fromEntries(Object.entries(props).filter(([key]) => !transitionDirectivesToCopyOnIsland.includes(key)));
+	for (const sym of Object.getOwnPropertySymbols(props)) {
+		res[sym] = props[sym];
+	}
+	return res;
+}
```

Comment:
If we move this instead `extractDirectives`, I think we can do the check at here directly? https://github.com/withastro/astro/blob/eaed844ea8f2f52e0c9caa40bb3ec7377e10595f/packages/astro/src/runtime/server/hydration.ts#L97-L99

Using `fromEntries` and `entries` has a higher perf cost.

Reply:
Yes.